### DISCUSSION TITLE
Is it possible to send Cmd-{/} through to tmux?

### DISCUSSION THREAD

**Discussion by smerrill** [kovidgoyal/kitty#6538] [Q&A] [answered]
Hey all - I love using Kitty as my Mac OS X terminal, but I'm having a single problem with it that prevents it from working as well as other terminal emulators.

On both Linux and Windows I am able to use the standard OS X keybinds for previous and next tab (`Alt-{` and/or `Alt-Shift-[` depending on how you spell it) to switch between tabs in tmux, as well as `Alt-t` to make a new tab.

I was able to have Kitty ignore `Cmd-t` and pass it directly through to tmux by setting it to a `no_op` with the following `kitty.conf` snippet:

```
map cmd+t no_op
```

This works great, and I am able to hit Cmd-t to make a new tmux pane using this stanza of tmux configuration.

```
bind-key -n M-t new-window -c "#{pane_current_path}"
```

So far, so good.

That said, no combination of `no_op`s will apparently pass through the default OS X key combinations to tmux. Here are the options I am trying in `kitty.conf`:

```
map cmd+} no_op
map cmd+{ no_op
map shift+cmd+] no_op
map shift+cmd+[ no_op
map shift+super+] no_op
map shift+super+[ no_op
```

Normally I only need this option in `tmux.conf` to receive that keypress on Linux or Windows:

```
bind-key -n 'M-{' previous-window
bind-key -n 'M-}' next-window
```

That said, I've also added the following in hopes of getting this to work:

```
bind-key -n 'M-S-[' previous-window
bind-key -n 'M-S-]' next-window
bind-key -n 'M-S-{' previous-window
bind-key -n 'M-S-}' next-window
```

But instead when I type `Cmd-{` / `Cmd-Shift-[` to try to go to the previous tab, tmux does nothing and I just get a `[` output in my terminal.

Any ideas where I could look for the answer to this? I'm not sure if this is something that I need to investigate on the kitty side, the tmux side, or both, but regardless, if I could delegate the normal OS X keybinds to make and move between tabs I'd be very happy.

**Comment by kovidgoyal** [comment, marked as answer]
I dont use tmux so I have no clue what key press it is expecting. You
can bind any key to send arbitrary bytes in kitty, using 
map whatever send_text all whatever 
in kitty.conf. Figure out what bytes tmux wants and use send_text
appropriately.

**Comment by smerrill** [comment]
Thank you very much for this software and this answer!

I ended up making the two keys send `Ctrl-a n` and `Ctrl-a p` respectively, which does the trick:

```
map shift+cmd+] send_text all \x01n
map shift+cmd+[ send_text all \x01p
```
